### PR TITLE
Add a separate flow of sync stages for sequencing

### PR DESCRIPTION
**Pull request by mandrigin**
also adding comments on sequencing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,10 @@ Please install:
 
 Using the Makefile command: `make build-libs` will install these for the relevant architecture.
 
+## sequencer (WIP)
+
+Enable Sequencer: `CDK_ERIGON_SEQUENCER=1 ./build/bin/cdk-erigon <flags>`
+q
```
Comment (hexoscott) [inline README.md]: Not sure the q should be here?

File: zk/stages/stage_sequence_execute.go [hunk 2]
```diff
@@ -93,7 +93,7 @@ type ExecuteBlockCfg struct {
 	zk        *ethconfig.Zk
 }
 
-func StageExecuteBlocksCfg(
+func StageSequenceBlocksCfg(
```
Comment (hexoscott) [inline zk/stages/stage_sequence_execute.go:96]: Should this be `StageSequenceExecutionCfg` instead with the returned type of `ExecuteBlockCfg` renamed to match also?

File: zk/stages/stage_sequence_execute.go [hunk 3]
```diff
@@ -394,17 +373,39 @@ Loop:
...
+		// write batches between last block and this if they exist
+		currentBatch, err := hermezDb.GetBatchNoByL2Block(blockNum)
+		if err != nil {
+			return err
+		}
+
+		/*
+		* SEQ: here in theory we should get GERs from the L1 and write them down before
+		* execution into a separate table.
+		 */
```
Comment (hexoscott) [inline zk/stages/stage_sequence_execute.go:394]: Could this be its own stage entirely?  To just keep a watch on GER events from the L1 so they're always ready ahead of time?

File: zk/stages/stages.go [hunk 4]
```diff
@@ -58,6 +55,8 @@ func DefaultZkSequencerStages(
 				return stagedsync.PruneBlockHashStage(p, tx, blockHashCfg, ctx)
 			},
 		},
+		/* TODO: here should be some stage of getting GERs from the L1 and writing to the DB for future batches
+		 */
```
Comment (hexoscott) [inline zk/stages/stages.go:46]: Matches in with the comment above about this being its own stage

File: zk/stages/stages.go [hunk 5]
```diff
@@ -58,6 +55,8 @@ func DefaultZkSequencerStages(
 				return stagedsync.PruneBlockHashStage(p, tx, blockHashCfg, ctx)
 			},
 		},
```
Comment (hexoscott) [inline zk/stages/stages.go]: If we're the only sequencer in operation would this not be covered during execution/sequencing?

**Review by hexoscott — COMMENTED** [review]
Looks good, I'm unsure around the unwind order and perhaps some unneccessary stages if we're running as a sequencer but that could be my lack of understanding.

File: zk/stages/stages.go (hunk 5 above)
Comment (hexoscott) [reply zk/stages/stages.go]: Not clear in the UI, but this is targetted at the block hashes stage

File: zk/stages/stage_sequence_execute.go (hunk 3 above)
Comment (mandrigin) [reply zk/stages/stage_sequence_execute.go:394]: yeah, good point, we need some L1 GER stage

File: zk/stages/stages.go (hunk 5 above)
Comment (mandrigin) [reply zk/stages/stages.go]: yeah, yeah, this will be during the execution, good catch

Comment (mandrigin) [reply zk/stages/stages.go]: removed that stage

File: zk/stages/stage_sequence_execute.go (hunk 2 above)
Comment (mandrigin) [reply zk/stages/stage_sequence_execute.go:96]: renamed

**Review by hexoscott — APPROVED** [review]

**Pull request merged**